tests/test_thread.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+import threading
+
+from pybind11_tests import thread as m
+
+
+def test_implicit_conversion():
+    def loop(count):
+        for i in range(count):
+            m.test(i)
+
+    a = threading.Thread(target=loop, args=(10,))
+    b = threading.Thread(target=loop, args=(10,))
+    c = threading.Thread(target=loop, args=(10,))
+    for x in [a, b, c]:
+        x.start()
+    for x in [c, b, a]:
+        x.join()
+
+
+def test_implicit_conversion_no_gil():
```

Comment:
Have you been able to see this test fail without the fix applied? I've not been able to on my machine locally. I think the move away from thread pool executor to just "raw" python threads means we'll have to do something extra to capture the exceptions. See https://stackoverflow.com/questions/2829329/catch-a-threads-exception-in-the-caller-thread for a few ideas.

Reply:
Fair point; fixed.